src/frontend/src/stream_fragmenter/mod.rs

```diff
@@ -196,11 +196,13 @@ impl StreamFragmenter {
             // TODO: Force singleton for TopN as a workaround. We should implement two phase TopN.
             NodeBody::TopN(_) => current_fragment.is_singleton = true,
 
-            NodeBody::Chain(ref node) => {
+            // FIXME: workaround for single-fragment mview on singleton upstream mview.
+            NodeBody::Chain(node) => {
                 // memorize table id for later use
                 state
                     .dependent_table_ids
                     .insert(TableId::new(node.table_id));
+                current_fragment.is_singleton = node.is_singleton;
```

Comment:
IIUC, the flag `node.is_singleton` for chain_node is alwasy true? Or it depends on the whether the upstream mview is singleton or not.

Reply:
> Or it depends on the whether the upstream mview is singleton or not.

✔️